service/src/main/java/tds/config/repositories/ConfigRepository.java

```diff
@@ -31,11 +35,36 @@
      *     {@code CommonDLL.selectIsOnByAuditObject} method.
      * </p>
      * <p>
-     *     Because these values will rarely change, they are strong candidates for caching.
+     *     Because the {@link ClientSystemFlag} values will rarely change, they are strong candidates for caching.
      * </p>
      *
-     * @param clientName The client name for which the {@code ClientSystemFlag} records should be fetched.
-     * @return A collection of {@code ClientSystemFlag} records for the specified client name.
+     * @param clientName The client name for which the {@link ClientSystemFlag} records should be fetched.
+     * @return A collection of {@link ClientSystemFlag} records for the specified client name.
      */
     List<ClientSystemFlag> getClientSystemFlags(String clientName);
+
+    /**
+     * Get the time limits from the {@code session.timelimits} view for the specified client name.
+     * <p>
+     *     Because the {@link TimeLimits} values will rarely change, they are strong candidates for caching.
+     * </p>
+     *
+     * @param clientName The client name for which the {@link TimeLimits} record should be fetched.
+     * @return An {@link Optional<TimeLimits>} representing the time limit configuration values; otherwise
+     *         {@code Optional.empty()}.
+     */
+    Optional<TimeLimits> getTimeLimits(String clientName);
+
+    /**
+     * Get the time limits from the {@code session.timelimits} view for the specified client name and assessment id.
+     * <p>
```

Comment:
Re-structured JavaDocs; minimized docs on methods & getters/setters.  Moved logic/description to the class level.  Removed almost all references to legacy code.